internal/pkg/cli/deploy/deploy.go

```diff
@@ -854,8 +858,9 @@ func (d *workloadDeployer) pushAddonsTemplateToS3Bucket(in *pushAddonsTemplateTo
 		}
 		return "", fmt.Errorf("retrieve addons template: %w", err)
 	}
-	reader := strings.NewReader(template)
-	url, err := in.uploader.Upload(d.resources.S3Bucket, fmt.Sprintf(deploy.AddonsCfnTemplateNameFormat, d.name), reader)
+	reader := strings.NewReader(tmpl)
+	artifactName := path.Join(d.name, fmt.Sprintf("%x.yml", sha256.Sum256([]byte(tmpl))))
```

Comment:
I wonder if we should move `sha256.Sum256` to `AddonsArtifactPath`. Same for `TemplateArtifactPath` because it seems like whenever we use them we always call `sha256.Sum256`

Reply:
I would prefer the key being determined by the client but I'm fine either way

Reply:
addressed

Reply:
But do we have an assumption that all the addons artifacts should do `sha256.Sum256`? Same for the env file and template uploads. If so maybe we would want
```go
// Env file upload
url, err := in.uploader.Upload(d.resources.S3Bucket, template.EnvFileArtifactPath(envFilePath, content), reader)

// Addons upload
url, err := in.uploader.Upload(d.resources.S3Bucket, template.AddonsArtifactPath(d.name), reader)

// Template upload
	url, err := cf.s3Client.Upload(bucket, template.TemplateArtifactPath(config.StackName()), reader)
```

it's ok if you don't want to change. Not a big one tho.

Reply:
I agree that we should most likely always upload with sha256 :) this way we can eliminate this bug altogether for future code changes